diff --git a/lib/module/StackRouter.js b/lib/module/StackRouter.js
index c1459d5918190a85b641710aabdcea37934aeeca..d8b6eb9cc4ed37d627c2c52058a2519195aae0bc 100644
--- a/lib/module/StackRouter.js
+++ b/lib/module/StackRouter.js
@@ -78,6 +78,12 @@ export function StackRouter(options) {
       routeParamList
     }) {
       const state = partialState;
+      if (!state) {
+        return router.getInitialState({
+          routeNames,
+          routeParamList
+        });
+      }
       if (state.stale === false) {
         return state;
       }
diff --git a/lib/module/TabRouter.js b/lib/module/TabRouter.js
index f3b26a991b60cec9729cd4851aa40a8e2e998479..a2b8eb9338c3e47b6aec25093e47a4e97c532145 100644
--- a/lib/module/TabRouter.js
+++ b/lib/module/TabRouter.js
@@ -119,6 +119,12 @@ export function TabRouter({
       routeParamList
     }) {
       const state = partialState;
+      if (!state) {
+        return router.getInitialState({
+          routeNames,
+          routeParamList
+        });
+      }
       if (state.stale === false) {
         return state;
       }
